### PR TITLE
Fix possible NPEx

### PR DESCRIPTION
**Pull request by pzygielo**
Due to possibility of `recommendedVersion` being null.

---

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/dependency/tree/TreeMojo.java [hunk 1]
```diff
@@ -479,9 +479,12 @@ public static boolean containsVersion( VersionRange allowedRange, ArtifactVersio
                     return true;
                 }
             }
+            return false;
+        }
+        else
+        {
+            // only singular versions ever have a recommendedVersion
+            return recommendedVersion.compareTo(theVersion) <= 0;
```
Comment (michael-o) [inline src/main/java/org/apache/maven/plugins/dependency/tree/TreeMojo.java:487]: Spaces missing

File: src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo_ContainsVersion.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.Collections;
+
+import static org.apache.maven.plugins.dependency.tree.TreeMojo.containsVersion;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests <code>TreeMojo.containsVersion</code>.
+ */
+public class TestTreeMojo_ContainsVersion extends TestCase
```
Comment (michael-o) [inline src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo_ContainsVersion.java:36]: Same here thoughout the class.

File: src/main/java/org/apache/maven/plugins/dependency/tree/TreeMojo.java (hunk 1 above)
Comment (pzygielo) [reply src/main/java/org/apache/maven/plugins/dependency/tree/TreeMojo.java]: Corrected. :cry: I don't know how it happened as it is clearly reported by `verify`.

File: src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo_ContainsVersion.java (hunk 2 above)
Comment (pzygielo) [reply src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo_ContainsVersion.java:36]: Spaces corrected, but still have to learn why `checkstyle-check` does not complain for these violations in `src/test`.

Comment (pzygielo) [reply src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo_ContainsVersion.java:36]: > still have to learn why `checkstyle-check` does not complain for these violations in `src/test`.

I guess due to [includeTestSourceDirectory](https://maven.apache.org/plugins/maven-checkstyle-plugin/check-mojo.html#includeTestSourceDirectory) having `Default: false`.

So [maven-parent](https://github.com/apache/maven-parent/blob/e4db8ae90caac88fb206c9003a93465172b9b86a/pom.xml#L984) adds `testSourceDirectory` but does not set `includeTestSourceDirectory`.

**Comment by pzygielo** [comment]
@michael-o may I ask for review, please?

**Comment by michael-o** [comment]
@pzygielo I will pick up your PRs this night. Currently busy. Thank you for your time and effort!

**Comment by pzygielo** [comment]
May I ask for review, please?

**Comment by michael-o** [comment]
Looking into...in an hour!

**Comment by pzygielo** [comment]
@michael-o thanks for checking and processing paper-work in JIRA.

**Comment by jira-importer** [comment]
Resolve #1220

**Pull request closed without merging**